### ISSUE TITLE
[Bug]: Unsupported image type in input when using input image

### ISSUE THREAD

**Issue by NaderTate** [lllyasviel/Fooocus#3502]
### Checklist

- [ ] The issue has not been resolved by following the [troubleshooting guide](https://github.com/lllyasviel/Fooocus/blob/main/troubleshoot.md)
- [ ] The issue exists on a clean installation of Fooocus
- [ ] The issue exists in the current version of Fooocus
- [ ] The issue has not been reported before recently
- [ ] The issue has been reported before but has not been fixed yet

### What happened?

Whether I'm using image prompt or inpaint/outpaint, I get the error: Unsupported image type in input.
Normal image generation from text works fine, but using any input image throws this error.
Others have reported the same issue on Reddit, so it's not just me.

### Steps to reproduce the problem

1. Run foocus.
2. Try to use inpaint on any image.

### What should have happened?

Just do the magic...

### What browsers do you use to access Fooocus?

Microsoft Edge

### Where are you running Fooocus?

Cloud (Google Colab)

### What operating system are you using?

Win 11

### Console logs

```Shell
Traceback (most recent call last):
  File "/content/Fooocus/modules/gradio_hijack.py", line 279, in preprocess
    im = processing_utils.decode_base64_to_image(x)
  File "/usr/local/lib/python3.10/dist-packages/gradio/processing_utils.py", line 59, in decode_base64_to_image
    img = Image.open(BytesIO(base64.b64decode(image_encoded)))
  File "/usr/local/lib/python3.10/dist-packages/PIL/Image.py", line 3283, in open
    rawmode = mode
PIL.UnidentifiedImageError: cannot identify image file <_io.BytesIO object at 0x782f4810f010>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/gradio/routes.py", line 488, in run_predict
    output = await app.get_blocks().process_api(
  File "/usr/local/lib/python3.10/dist-packages/gradio/blocks.py", line 1429, in process_api
    inputs = self.preprocess_data(fn_index, inputs, state)
  File "/usr/local/lib/python3.10/dist-packages/gradio/blocks.py", line 1239, in preprocess_data
    processed_input.append(block.preprocess(inputs[i]))
  File "/content/Fooocus/modules/gradio_hijack.py", line 281, in preprocess
    raise Error("Unsupported image type in input")
gradio.exceptions.Error: 'Unsupported image type in input'
```

### Additional information

I noticed gradio was down today, I don't know if this has anything to do with the issue

**Comment by mashb1t** [comment]
Can't reproduce, works fine for me. Can you please provide the full console log as well as the image you're using? (please check if after uploading and downloading from Github it still doesn't work for you)

**Comment by NaderTate** [comment]
I'm running it on colab, here's the code I always use:

!pip install pygit2==1.12.2
%cd /content
!git clone https://github.com/lllyasviel/Fooocus.git
%cd /content/Fooocus
!apt-get install aria2
!python entry_with_update.py --share --always-high-vram

Here's what I did:
1. after launching fooocus, I typed "cat" in the prompt, I didn't check the advanced tab or the input image tab or anything.
2. I got 2 cat images, I dragged one of them to the inpaint tab, marked over the cat's eyes, selected "improve details" and typed "detailed eyes" in the prompt.
Then I got the error.

Here are screenshots:

![Screenshot_1](https://github.com/user-attachments/assets/303d94bb-34aa-48f6-a552-86545e396c3a)
![Screenshot_2](https://github.com/user-attachments/assets/7363b6fe-125b-409c-a402-9ad2d52257df)

Here's the full console.log:

`Collecting pygit2==1.12.2
  Downloading pygit2-1.12.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.5 kB)
Requirement already satisfied: cffi>=1.9.1 in /usr/local/lib/python3.10/dist-packages (from pygit2==1.12.2) (1.17.0)
Requirement already satisfied: pycparser in /usr/local/lib/python3.10/dist-packages (from cffi>=1.9.1->pygit2==1.12.2) (2.22)
Downloading pygit2-1.12.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.9 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.9/4.9 MB 23.2 MB/s eta 0:00:00
Installing collected packages: pygit2
Successfully installed pygit2-1.12.2
/content
Cloning into 'Fooocus'...
remote: Enumerating objects: 6703, done.
remote: Counting objects: 100% (16/16), done.
remote: Compressing objects: 100% (13/13), done.
remote: Total 6703 (delta 3), reused 7 (delta 2), pack-reused 6687
Receiving objects: 100% (6703/6703), 33.25 MiB | 28.88 MiB/s, done.
Resolving deltas: 100% (3865/3865), done.
/content/Fooocus
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  libaria2-0 libc-ares2
The following NEW packages will be installed:
  aria2 libaria2-0 libc-ares2
0 upgraded, 3 newly installed, 0 to remove and 45 not upgraded.
Need to get 1,513 kB of archives.
After this operation, 5,441 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libc-ares2 amd64 1.18.1-1ubuntu0.22.04.3 [45.1 kB]
Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libaria2-0 amd64 1.36.0-1 [1,086 kB]
Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 aria2 amd64 1.36.0-1 [381 kB]
Fetched 1,513 kB in 0s (5,232 kB/s)
Selecting previously unselected package libc-ares2:amd64.
(Reading database ... 123594 files and directories currently installed.)
Preparing to unpack .../libc-ares2_1.18.1-1ubuntu0.22.04.3_amd64.deb ...
Unpacking libc-ares2:amd64 (1.18.1-1ubuntu0.22.04.3) ...
Selecting previously unselected package libaria2-0:amd64.
Preparing to unpack .../libaria2-0_1.36.0-1_amd64.deb ...
Unpacking libaria2-0:amd64 (1.36.0-1) ...
Selecting previously unselected package aria2.
Preparing to unpack .../aria2_1.36.0-1_amd64.deb ...
Unpacking aria2 (1.36.0-1) ...
Setting up libc-ares2:amd64 (1.18.1-1ubuntu0.22.04.3) ...
Setting up libaria2-0:amd64 (1.36.0-1) ...
Setting up aria2 (1.36.0-1) ...
Processing triggers for man-db (2.10.2-1) ...
Processing triggers for libc-bin (2.35-0ubuntu3.4) ...
/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link

/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link

/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link

/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link

/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link

/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link

/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link

/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link

/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link

Already up-to-date
Update succeeded.
[System ARGV] ['entry_with_update.py', '--share', '--always-high-vram']
Python 3.10.12 (main, Jul 29 2024, 16:56:48) [GCC 11.4.0]
Fooocus version: 2.5.4
Error checking version for torchsde: No package metadata was found for torchsde
Installing requirements
[Cleanup] Attempting to delete content of temp dir /tmp/fooocus
[Cleanup] Cleanup successful
Downloading: "https://huggingface.co/lllyasviel/misc/resolve/main/xlvaeapp.pth" to /content/Fooocus/models/vae_approx/xlvaeapp.pth

100% 209k/209k [00:00<00:00, 6.96MB/s]
Downloading: "https://huggingface.co/lllyasviel/misc/resolve/main/vaeapp_sd15.pt" to /content/Fooocus/models/vae_approx/vaeapp_sd15.pth

100% 209k/209k [00:00<00:00, 5.29MB/s]
Downloading: "https://huggingface.co/mashb1t/misc/resolve/main/xl-to-v1_interposer-v4.0.safetensors" to /content/Fooocus/models/vae_approx/xl-to-v1_interposer-v4.0.safetensors

100% 5.40M/5.40M [00:00<00:00, 64.4MB/s]
Downloading: "https://huggingface.co/lllyasviel/misc/resolve/main/fooocus_expansion.bin" to /content/Fooocus/models/prompt_expansion/fooocus_expansion/pytorch_model.bin

100% 335M/335M [00:01<00:00, 241MB/s]
Downloading: "https://huggingface.co/lllyasviel/fav_models/resolve/main/fav/juggernautXL_v8Rundiffusion.safetensors" to /content/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors

100% 6.62G/6.62G [00:50<00:00, 141MB/s] 
Downloading: "https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/resolve/main/sd_xl_offset_example-lora_1.0.safetensors" to /content/Fooocus/models/loras/sd_xl_offset_example-lora_1.0.safetensors

100% 47.3M/47.3M [00:00<00:00, 163MB/s]
Total VRAM 15102 MB, total RAM 12979 MB
Set vram state to: HIGH_VRAM
Always offload VRAM
Device: cuda:0 Tesla T4 : native
VAE dtype: torch.float32
Using pytorch cross attention
Refiner unloaded.
IMPORTANT: You are using gradio version 3.41.2, however version 4.29.0 is available, please upgrade.
--------
Running on local URL:  http://127.0.0.1:7865/
model_type EPS
UNet ADM Dimension 2816
Running on public URL: https://821eff3d9b75360ee6.gradio.live/

This share link expires in 72 hours. For free permanent hosting and GPU upgrades, run `gradio deploy` from Terminal to deploy to Spaces (https://huggingface.co/spaces)
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra {'cond_stage_model.clip_l.text_projection', 'cond_stage_model.clip_l.logit_scale'}
left over keys: dict_keys(['cond_stage_model.clip_l.transformer.text_model.embeddings.position_ids'])
loaded straight to GPU
Requested to load SDXL
Loading 1 new model
Base model loaded: /content/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors
VAE loaded: None
Request to load LoRAs [('sd_xl_offset_example-lora_1.0.safetensors', 0.1)] for model [/content/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors].
Loaded LoRA [/content/Fooocus/models/loras/sd_xl_offset_example-lora_1.0.safetensors] for UNet [/content/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 788 keys at weight 0.1.
Fooocus V2 Expansion: Vocab with 642 words.
Fooocus Expansion engine loaded for cuda:0, use_fp16 = True.
Requested to load SDXLClipModel
Requested to load GPT2LMHeadModel
Loading 2 new models
[Fooocus Model Management] Moving model(s) has taken 0.76 seconds
2024-08-11 21:05:01.436804: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:485] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-08-11 21:05:01.767727: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:8454] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-08-11 21:05:01.853417: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1452] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-08-11 21:05:02.384285: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-08-11 21:05:04.594615: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
Started worker with PID 874
App started successful. Use the app with http://127.0.0.1:7865/ or 127.0.0.1:7865 or https://821eff3d9b75360ee6.gradio.live/
[Parameters] Adaptive CFG = 7
[Parameters] CLIP Skip = 2
[Parameters] Sharpness = 2
[Parameters] ControlNet Softness = 0.25
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] Seed = 885801788671351321
[Parameters] CFG = 4
[Fooocus] Loading control models ...
[Parameters] Sampler = dpmpp_2m_sde_gpu - karras
[Parameters] Steps = 30 - 15
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Refiner unloaded.
[Fooocus] Processing prompts ...
[Fooocus] Preparing Fooocus text #1 ...
[Prompt Expansion] cat, dramatic cinematic light, detailed, warm colors, highly detail, intricate, elegant, epic, creative, colorful, surreal, beautiful, sharp focus, great composition,, clear professional atmosphere, dynamic, lively, amazing, aesthetic, very inspirational, cute, confident, color, awesome, best, whole, full, pretty, perfect, background, sunny, magic, deep
[Fooocus] Preparing Fooocus text #2 ...
[Prompt Expansion] cat, bright colors, highly detailed, intricate, sharp focus, cinematic, joyful, beautiful, expressive, symmetry, illuminated glowing light, vivid color, coherent, cute, lovely, epic, fine detail, full, pretty, perfect, colorful, background, complex, amazing composition, professional, elegant, atmosphere, new, great, focused, creative,, positive, unique
[Fooocus] Encoding positive #1 ...
[Fooocus Model Management] Moving model(s) has taken 0.13 seconds
[Fooocus] Encoding positive #2 ...
[Fooocus] Encoding negative #1 ...
[Fooocus] Encoding negative #2 ...
[Parameters] Denoising Strength = 1.0
[Parameters] Initial Latent shape: Image Space (896, 1152)
Preparation time: 4.34 seconds
Using karras scheduler.
[Fooocus] Preparing task 1/2 ...
[Sampler] refiner_swap_method = joint
[Sampler] sigma_min = 0.0291671771556139, sigma_max = 14.614643096923828
Requested to load SDXL
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 2.20 seconds
100% 30/30 [00:27<00:00,  1.09it/s]
Requested to load AutoencoderKL
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 0.37 seconds
[Fooocus] Saving image 1/2 to system ...
Image generated with private log at: /content/Fooocus/outputs/2024-08-11/log.html
Generating and saving time: 33.09 seconds
[Fooocus] Preparing task 2/2 ...
[Sampler] refiner_swap_method = joint
[Sampler] sigma_min = 0.0291671771556139, sigma_max = 14.614643096923828
Requested to load SDXL
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 0.74 seconds
100% 30/30 [00:28<00:00,  1.04it/s]
Requested to load AutoencoderKL
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 0.29 seconds
[Fooocus] Saving image 2/2 to system ...
Image generated with private log at: /content/Fooocus/outputs/2024-08-11/log.html
Generating and saving time: 32.80 seconds
[Enhance] Skipping, preconditions aren't met
Processing time (total): 65.89 seconds
Requested to load SDXLClipModel
Requested to load GPT2LMHeadModel
Loading 2 new models
Total time: 96.09 seconds
[Fooocus Model Management] Moving model(s) has taken 0.90 seconds
Traceback (most recent call last):
  File "/content/Fooocus/modules/gradio_hijack.py", line 279, in preprocess
    im = processing_utils.decode_base64_to_image(x)
  File "/usr/local/lib/python3.10/dist-packages/gradio/processing_utils.py", line 59, in decode_base64_to_image
    img = Image.open(BytesIO(base64.b64decode(image_encoded)))
  File "/usr/local/lib/python3.10/dist-packages/PIL/Image.py", line 3283, in open
    rawmode = mode
PIL.UnidentifiedImageError: cannot identify image file <_io.BytesIO object at 0x7936e05d6070>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/gradio/routes.py", line 488, in run_predict
    output = await app.get_blocks().process_api(
  File "/usr/local/lib/python3.10/dist-packages/gradio/blocks.py", line 1429, in process_api
    inputs = self.preprocess_data(fn_index, inputs, state)
  File "/usr/local/lib/python3.10/dist-packages/gradio/blocks.py", line 1239, in preprocess_data
    processed_input.append(block.preprocess(inputs[i]))
  File "/content/Fooocus/modules/gradio_hijack.py", line 281, in preprocess
    raise Error("Unsupported image type in input")
gradio.exceptions.Error: 'Unsupported image type in input'
[Parameters] Adaptive CFG = 7
[Parameters] CLIP Skip = 2
[Parameters] Sharpness = 2
[Parameters] ControlNet Softness = 0.25
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] Seed = 885801788671351321
[Parameters] CFG = 4
[Fooocus] Loading control models ...
[Parameters] Sampler = dpmpp_2m_sde_gpu - karras
[Parameters] Steps = 30 - 15
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Refiner unloaded.
[Fooocus] Processing prompts ...
[Fooocus] Encoding positive #1 ...
[Fooocus Model Management] Moving model(s) has taken 0.15 seconds
[Fooocus] Encoding positive #2 ...
[Fooocus] Encoding negative #1 ...
[Fooocus] Encoding negative #2 ...
[Parameters] Denoising Strength = 1.0
[Parameters] Initial Latent shape: Image Space (896, 1152)
Preparation time: 0.62 seconds
Using karras scheduler.
[Fooocus] Preparing task 1/2 ...
[Sampler] refiner_swap_method = joint
[Sampler] sigma_min = 0.0291671771556139, sigma_max = 14.614643096923828
Requested to load SDXL
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 1.79 seconds
100% 30/30 [00:29<00:00,  1.01it/s]
Requested to load AutoencoderKL
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 0.35 seconds
[Fooocus] Saving image 1/2 to system ...
Image generated with private log at: /content/Fooocus/outputs/2024-08-11/log.html
Generating and saving time: 34.90 seconds
[Fooocus] Preparing task 2/2 ...
[Sampler] refiner_swap_method = joint
[Sampler] sigma_min = 0.0291671771556139, sigma_max = 14.614643096923828
Requested to load SDXL
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 0.69 seconds
100% 30/30 [00:28<00:00,  1.07it/s]
Requested to load AutoencoderKL
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 0.31 seconds
[Fooocus] Saving image 2/2 to system ...
Image generated with private log at: /content/Fooocus/outputs/2024-08-11/log.html
Generating and saving time: 31.74 seconds
[Enhance] Skipping, preconditions aren't met
Processing time (total): 66.64 seconds
Requested to load SDXLClipModel
Requested to load GPT2LMHeadModel
Loading 2 new models
Total time: 67.34 seconds
[Fooocus Model Management] Moving model(s) has taken 0.86 seconds`

**Comment by NaderTate** [comment]
Also I noticed that when I selected the cat's eyes and clicked generate, after getting the error, it started generating 2 **new** images, and they were added to the previous images, so after I clicked generate twice I ended up with 6 images, which is weird...
![image](https://github.com/user-attachments/assets/d809fe45-b6bc-4bbf-a444-58f164b0c54d)

**Comment by mashb1t** [comment]
This only affects Colab and has nothing to do with the latest update v2.5.4. I don't know what has changed on their side but stopping and starting Fooocus on Colab again after the initial start solves the issue.

Steps:
1. start Fooocus in Colab
2. wait until the share link is shown - this would also be the step where the error is snown
3. click stop
4. click play, wait until the share link is shown
5. you can now use it without issues

Please check if this also works for you.

**Comment by marceloflix** [comment]
@mashb1t Had the same issue but not anymore after following the steps, thanks!

**Comment by mashb1t** [comment]
It actually had to do with an import statement being at the wrong position. Fixed in https://github.com/lllyasviel/Fooocus/pull/3506, merged to main and now available in v2.5.5.
Sorry for the inconvenience.
